src/development/LocalTest/Controllers/Storage/InstancesController.cs

```diff
@@ -422,6 +422,54 @@ public async Task<ActionResult<Instance>> Delete(int instanceOwnerPartyId, Guid
             return Ok(updatedInstance);
         }
 
+        /// <summary>
+        /// Update instance read status.
+        /// </summary>
+        /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+        /// <param name="instanceGuid">The id of the instance to confirm as complete.</param>
+        /// <param name="status">The updated read status.</param>
+        /// <returns>Returns the updated instance.</returns>        
+        [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_READ)]
+        [HttpPut("{instanceOwnerPartyId:int}/{instanceGuid:guid}/readstatus")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Produces("application/json")]
+        public async Task<ActionResult<Instance>> UpdateReadStatus(
+          [FromRoute] int instanceOwnerPartyId,
+          [FromRoute] Guid instanceGuid,
+          [FromQuery] string status)
+        {
+            string instanceId = $"{instanceOwnerPartyId}/{instanceGuid}";
+
+            Instance instance = await _instanceRepository.GetOne(instanceId, instanceOwnerPartyId);
+
+            if (!Enum.TryParse(status, true, out ReadStatus newStatus))
+            {
+                return BadRequest($"Invalid read status: {status}. Accepted types include: {string.Join(", ", Enum.GetNames(typeof(ReadStatus)))}");
+            }
```

Comment:
This if block can be moved to the top of the method. No need to retrieve the instance if this has failed.

Reply:
good point. Done :)